deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -38,6 +38,14 @@
 partitioned_param_data_shape = [0]
 zero_init_context = 0
 top_level_context = None
+_cls_excluded_from_partitioning = set()
```

Comment:
Now that deepspeed supports multiple models instantiation won't introducing a global state like this potentially lead to problems?

May be  `_cls_excluded_from_partitioning` should be cleared at the end of Init

It's hard to say if the user will want the same behavior for a sub-module with the same name in both instances or not.

Reply:
This is intended for internal use inside DeepSpeed, to avoid having to explicitly shut down the Init context inside deepspeed.initialize. If we want to let users exclude their own classes from partitioning I think we need to rethink this. Do you have a usecase in mind?

Reply:
I was just mentioning this since DS started to support multiple concurrent models, so it's hard to tell what users will try to do. That's why I'm voicing a concern about a global variable that could impact a complex setup.

I don't have a concrete case in mind right now, other than someone trying to train 2 models where one submodule is shared.

Reply:
Agreed. @tjruwase I believe you have experience with this kind of use case?

Reply:
@eisene, yes, I have being trying to eliminate global variables from this file to support partitioning of multiple models, e.g., RLHF training.